crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -327,13 +330,63 @@ async fn execute_ics20_transfer<S: StateWriteExt>(
     // if the asset is prefixed with `ibc`, the rest of the denomination string is the asset ID,
     // so we need to look up the full trace from storage.
     // see https://github.com/cosmos/ibc-go/blob/main/docs/architecture/adr-001-coin-source-tracing.md#decision
-    if denom.prefix_is("ibc") {
+    if denom.prefix().starts_with("ibc") {
         denom = state
             .get_ibc_asset(denom.id())
             .await
             .context("failed to get denom trace from asset id")?;
     }
 
+    // check if the recipient is a bridge account; if so,
+    // ensure that the packet memo field is set, as the value in it
+    // should be the rollup destination address.
+    //
+    // also, ensure that the asset ID being transferred
+    // to it is allowed.
+    let maybe_recipient_rollup_id = state
+        .get_bridge_account_rollup_id(&recipient)
+        .await
+        .context("failed to get bridge account rollup ID from state")?;
+    let is_bridge_lock = maybe_recipient_rollup_id.is_some();
+
+    // note: bridge accounts *are* allowed to do ICS20 withdrawals,
```

Comment:
With the addition of this code, this method is growing very long and is hard to follow. Can you factor this out into self-standing functions?